staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -178,9 +178,6 @@ func ValidateManagedFields(fieldsList []metav1.ManagedFieldsEntry, fldPath *fiel
 		default:
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("operation"), fields.Operation, "must be `Apply` or `Update`"))
 		}
-		if fields.FieldsType != "FieldsV1" {
```

Comment:
why do we need to drop this validation? by the time we get here, wouldn't we expect to have replaced the managed fields with the generated ones if needed?

Reply:
rather than dropping this, I'd suggest changing it to making it to `if len(fields.FieldsType) > 0 && fields.FieldsType != "FieldsV1" {`

That would be a safeguard against garbage in the field, and would keep compatibility with older clients that drop the field value